rfcs/text/0352-graph-type-system.md

```diff
@@ -0,0 +1,2659 @@
+
+The `Contrived Property` Property Type could define its value as being _either_ an instance of the `Number` Data Type, _or_ an object which has a list with a maximum of 4 `Foo` property values
+
+```json
+{
+  "kind": "propertyType",
+  "$id": "https://blockprotocol.org/types/@alice/property-type/contrived-property",
+  "name": "Contrived Property",
+  "oneOf": [
+    {
+      "$ref": "https://blockprotocol.org/types/@blockprotocol/data-type/number"
+    },
+    {
+      "type": "array",
+      "items": {
+        "oneOf": [
+          {
+            "$ref": "https://blockprotocol.org/types/@blockprotocol/data-type/number"
+          }
+        ]
+      },
+      "maxItems": 4
+    }
+  ]
+}
+```
+
+Sample data when used in (simplified view of) an Entity
+
+Assuming that `Foo` accepts an instance of the `Text` Data Type
+
+```json
+{
+  "https://blockprotocol.org/types/@alice/property-type/contrived-property": 32
+}
+```
+
+or
+
+```json
+{
+  "https://blockprotocol.org/types/@alice/property-type/contrived-property": {
+    "https://blockprotocol.org/types/@alice/property-type/foo": "something here"
+  },
+  ...
+}
+```
+
+## Link Types
+
+A **Link Type** is a JSON schema that satisfies the following JSON meta-schema:
+
+```json
+{
+  "$id": "https://blockprotocol.org/type-system/0.2/schema/meta/link-type",
+  "description": "Specifies the structure of a Link Type",
+  "properties": {
+    "kind": {
+      "const": "linkType"
+    },
+    "$id": {
+      "type": "string",
+      "format": "uri"
+    },
+    "name": { "type": "string" },
+    "description": { "type": "string" },
+    "relatedKeywords": {
+      "type": "array",
+      "items": { "type": "string" }
+    }
+  },
+  "additionalProperties": false,
+  "required": ["kind", "$id", "name", "description"]
+}
+```
+
+> 💭 As the next section (Entity Types) will outline, there's an additional step to quantifying the types of things on either end of the relationship, as such the descriptions **should** be very general and most of the time not contain specific words about types (such as person).
+
+**Example 1** - `Owns`
+
+```json
+{
+  "kind": "linkType",
+  "$id": "https://blockprotocol.org/types/@alice/link-type/owns",
+  "name": "Owns",
+  "description": "Have (something) as one's own; possess"
+}
+```
+
+**Example 2** - `Submitted By`
+
+```json
+{
+  "kind": "linkType",
+  "$id": "https://blockprotocol.org/types/@alice/link-type/submitted-by",
+  "name": "Submitted By",
+  "description": "Suggested, proposed, or presented by"
+}
+```
+
+## Entity Types
+
+An **Entity Type** is a JSON schema that satisfies the following JSON meta-schema:
+
+```json
+{
+  "$id": "https://blockprotocol.org/type-system/0.2/schema/meta/entity-type",
+  "description": "Specifies the structure of an Entity Type",
+  "type": "object",
+  "properties": {
+    "kind": {
+      "const": "entityType"
+    },
+    "$id": {
+      "type": "string",
+      "format": "uri"
+    },
+    "name": { "type": "string" },
```

Comment:
We started off with `title` but I'm not 100% convinced yet that it is the same usage, perhaps it is.

The `title` property is the name of the **schema**, it's specifically about what the schema does, in the same way that we have to be careful about our usages of `description` because describing the type _might not be_ the same as describing the schema of the type.

If we can convince ourselves that they're the same then I agree we should use `title`.

Reply:
Here is the [latest JSON Schema definition](https://json-schema.org/draft/2020-12/json-schema-validation.html#name-title-and-description) for `title` and `description`:
> Both of these keywords can be used to decorate a user interface with information about the data produced by this user interface. A title will preferably be short, whereas a description will provide explanation about the purpose of the instance described by this schema.

Seems wide enough to accommodate what we want, but I haven't been in the discussion where you switched away from `title` so may be some nuance to `name` I'm missing.

Reply:
I haven't come up with any good reason for us to not use `title` so we'll switch. Thanks for the suggestion!

Reply:
Addressed in https://github.com/blockprotocol/blockprotocol/pull/352/commits/644bfc128e2b0c55e907edff429e6dfa3843dab3, should be updated everywhere where name is mentioned in the context of the schema `name`